### PR TITLE
Generate vesting spec from csv

### PR DESCRIPTION
**Pull request by frankli-dev**
`node ./scripts/generate_vesting_spec.js ./vesting.csv`

### CODE REVIEW

**Review by simonsso — CHANGES_REQUESTED** [review]
Looks ok, but I would like this script in a separate repo

File: scripts/generate_vesting_spec.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+`;
+      investors.map((inv) => {
+        balances += `   (hex!["${inv.address}"].into(), ${inv.total}),
+`;
+        let value = inv.amounts[0];
+        let stIndex = 0;
+        for (let i = 1; i < inv.amounts.length; i++) {
+          if (value !== inv.amounts[i]) {
+            vesting += `    (hex!["${inv.address}"].into(), ${getBlockNumber(dates[stIndex])}, ${
+              (30 * 24 * 60 * 60) / blockDuration
```
Comment (simonsso) [inline scripts/generate_vesting_spec.js:84]: This should be (365.25 days * blocks/day )/4 as we discussed yesterday

**Comment by frankli-dev** [comment]
> Looks ok, but I would like this script in a separate repo

okay. can you create a new repo?

File: scripts/generate_vesting_spec.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  const dates = [
+    '16/10/2021',
+    '16/01/2022',
+    '16/04/2022',
+    '16/07/2022',
+    '16/10/2022',
+    '16/01/2023',
+    '16/04/2023',
+    '16/07/2023',
+    '17/10/2023'
```
Comment (simonsso) [inline scripts/generate_vesting_spec.js:31]: Let's only support the first 45 which (with 3 dates) are the same the last five we code manually

File: scripts/generate_vesting_spec.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        let value = inv.amounts[0];
+        let stIndex = 0;
+        for (let i = 1; i < inv.amounts.length; i++) {
+          if (value !== inv.amounts[i]) {
+            vesting += `    (hex!["${inv.address}"].into(), ${getBlockNumber(dates[stIndex])}, ${
+              (30 * 24 * 60 * 60) / blockDuration
+            }, ${i - stIndex}, ${value}),
+`;
+            stIndex = i;
+            value = inv.amounts[i];
```
Comment (simonsso) [inline scripts/generate_vesting_spec.js:88]: Convert value to token by mutliplication with 10^12 (if there is an overflow problem multiply with 10 and add remaining 0 in print)

**Review by simonsso — CHANGES_REQUESTED** [review]

File: scripts/generate_vesting_spec.js (hunk 3 above)
Comment (frankli-dev) [reply scripts/generate_vesting_spec.js:88]: okay

File: scripts/generate_vesting_spec.js (hunk 2 above)
Comment (frankli-dev) [reply scripts/generate_vesting_spec.js:31]: okay

**Comment by Gauthamastro** [comment]
Implemented in a separate branch

**Pull request closed without merging**